presto-main/src/test/java/io/prestosql/operator/scalar/TestDateTimeFunctions.java

```diff
@@ -265,6 +265,16 @@ public void testFromUnixTimeWithTimeZone()
         assertFunction(format("from_unixtime(7200, '%s')", zoneId), TIMESTAMP_WITH_TIME_ZONE, toTimestampWithTimeZone(expected));
     }
 
+    @Test
+    public void testFromUnixTimeNanos()
+    {
+        assertFunction("from_unixtime_nanos(1234567890123456789)", TIMESTAMP_NANOS, sqlTimestampOf(9, 1234567890L, 123456789));
+        assertFunction("from_unixtime_nanos(1234567890000000000)", TIMESTAMP_NANOS, sqlTimestampOf(9, 1234567890L, 0));
+        assertFunction("from_unixtime_nanos(999999999)", TIMESTAMP_NANOS, sqlTimestampOf(9, 0, 999999999));
+        assertFunction("from_unixtime_nanos(DECIMAL '12345678900123456789')", TIMESTAMP_NANOS, sqlTimestampOf(9, 12345678900L, 123456789));
+        assertFunction("from_unixtime_nanos(DECIMAL '12345678900123456789.999')", TIMESTAMP_NANOS, sqlTimestampOf(9, 12345678900L, 123456789));
```

Comment:
Shouldn't we round in such case? @findepi @martint ?

Reply:
It should either round or fail, but truncation is inconsistent with datetime semantics in the engine.

Reply:
ok I'll change it to fail